MBBSEmu.Tests/ExportedModules/Majorbbs/strtol_Tests.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace MBBSEmu.Tests.ExportedModules.Majorbbs
+{
+    public class strtol_Tests : ExportedModuleTestBase
+    {
+        private const int STRTOL_ORDINAL = 586;
+
+        [Theory]
+        [InlineData("123456 Hello some text", 7, 10, 123456)]
```

Comment:
how is `ushortPtr` being used here? We should be asserting on the returned pointer that it points to the proper index. I guess for the first value of "123456", we should assert that the returned pointer is src + 6, since it reads 6 characters and should point the end to the space between "6" and "Hello"

The next test should assert on a value of 7, "78" would be 2 ahead of course.

Reply:
We'll also want to have a `[Fact]` test that sets the `endptr` to `null` to verify it doesn't try to write to the `null` value.

Reply:
After looking at the `endptr` stuff is implemented, I'm pretty sure it's not implemented correctly. The tests will validate this pretty easily.

Reply:
@paladine There is a + 1 in the ordinal that is making it 1 off from your logic above -- before I remove can you take a look now? 

I have it passing all tests and also properly getting the pointer afterwards. I also added the NULL fact test and added a couple more things to the ordinal to catch the leading blanks, blank string, and no numbers at all.

Let me know if i should remove the +1 from the ordinal (see below where it sets the suffix pointer):

```
if (suffixPointer != IntPtr16.Empty)
                Module.Memory.SetPointer(suffixPointer,
                    new IntPtr16(stringPointer.Segment, (ushort)(stringPointer.Offset + longToParseLength + 1)));
```